core/src/main/scala-2.12/cats/instances/stream.scala

```diff
@@ -16,6 +16,8 @@ trait StreamInstances extends cats.kernel.instances.StreamInstances {
 
       def combineK[A](x: Stream[A], y: Stream[A]): Stream[A] = x #::: y
 
+      override def prependK[A](a: A, fa: Stream[A]): Stream[A] = a #:: fa
```

Comment:
For `LazyList` you overrode both `prependK` and `appendK`, why only `prependK` for `Stream`, are the implementations different?

Reply:
Right, this is because `appendK` for `LazyList` utilizes `LazyList#appended` which is optimized for `LazyList`. But `appended`/`prepended` methods were first introduced in Scala 2.13 and `Stream` in Scala 2.12 does not have such methods yet. Therefore seems the best we can do here is to call
```scala
def appendK[A](fa: Stream[A], a: A) = fa #::: Stream(a)
```
but it is exactly what `combineK` (along with `pure`) already does.

Reply:
On the second though, `Stream` in Scala 2.13 does have the `prepended`/`appended` methods, but as I can see they're simply come from `Seq` and aren't optimized for `Stream` (unlike those in `LazyList`).

Reply:
Makes sense, thanks for explaining!